src/pages/index.js

```diff
@@ -1,62 +1,169 @@
 import "./index.css";
+import { data } from "autoprefixer";
+
+export const api = new Api({
+  baseUrl: "https://around.nomoreparties.co/v1/group-12",
+  headers: {
+    authorization: "cd56466a-c69a-4082-86d8-efb553341f31",
+    "Content-Type": "application/json",
+  },
+});
 
 // profile buttons
 const profileAddBtn = document.querySelector("#profile-add-button");
 const profileEditBtn = document.querySelector("#profile-edit-btn");
+const profileImgEditBtn = document.querySelector("#edit-avatar-btn");
 // forms
 const editForm = document.querySelector("#profile-edit-modal");
 const addForm = document.querySelector("#profile-add-modal");
+const profileImageForm = document.querySelector("#edit-profile-img-modal");
 // form inputs
 const profileNameInput = editForm.querySelector("#profile-name-input");
 const profileDescriptionInput = editForm.querySelector(
   "#profile-description-input"
 );
 
-const userInfo = new UserInfo("#profile-name", "#profile-des");
+const userInfo = new UserInfo(
+  "#profile-name",
+  "#profile-des",
+  "#profile-image"
+);
+let userID;
+api
+  .getUserinfo()
+  .then((data) => {
+    const { name, about, avatar, _id } = data;
+    userID = _id;
+    userInfo.setUserInfo(name, about);
+    userInfo.setUserProfileImage(avatar);
+  })
+  .catch((error) => {
+    console.error("Error fetching data:", error);
+  });
 
 // functions
 function handleProfileFormSubmit(data) {
-  userInfo.setUserInfo(data);
-  profileEditPopup.close();
+  const { name, about } = data;
+  api
+    .updateProfile(name, about)
+    .then(() => {
+      profileEditPopup.editSubmitBtn();
+    })
+    .then(() => {
+      userInfo.setUserInfo(name, about);
+      profileEditPopup.close();
+      profileEditPopup.resetSubmitBtn();
+    })
+    .catch((error) => {
+      console.error("Error fetching data:", error);
+    });
 }
 
 function handleCardFormSubmit(data) {
-  const name = data.title;
-  const link = data.imageUrl;
-  const card = createCard({ name, link });
-  cardSection.prependItem(card);
-  profileAddPopup.close();
+  api.postCard(data).then((data) => {
+    const card = createCard(data);
+    cardSection.prependItem(card);
+    profileAddPopup.close();
+    profileAddPopup.resetSubmitBtn();
+  });
+}
+
+function handleDeleteIcon(cardId, card) {
+  deleteCardPopup.open(cardId);
+  deleteCardPopup.setSubmitAction(() => {
+    api
+      .deleteCard(cardId)
+      .then(() => {
+        deleteCardPopup.close();
+        card.remove();
+      })
+      .catch((err) => {
+        console.error(err);
+      });
+  });
+}
+
+function handleProfileImgFormSubmit(data) {
+  api
+    .updateProfileImg(data.link)
+    .then(() => {
+      userInfo.setUserProfileImage(data.link);
+      profileImagePopup.close();
+    })
+    .catch((error) => {
+      console.error("Error fetching data:", error);
+    });
 }
 
 function handleCardClick(name, link) {
   previewImagePopup.open(name, link);
 }
+function addLike(cardId, likeCounter) {
```

Comment:
**Might be better**


To have access to any method of the class in `index.js` you can pass `this` to the call:

```js

        this._likeButton.addEventListener('click', () => {
            this._handleLike(this);
        });


```

And now in `index.js` you can call the methods like this:


```js


function addLike(cardInstance) {
        api.likeCard(cardInstance.getId())
            .then((res) => {
               cardInstance.addLike(res)  // inside it you'll toggle the heart color
            })
            .catch((err) => {
                console.error(err);
            });
}


```

The same thing is with deleting cards